persistence/transaction_test.go

```diff
@@ -63,10 +65,61 @@ var _ = Describe("func WithTransaction", func() {
 		Expect(items).NotTo(BeEmpty())
 	})
 
+	It("returns the transaction result", func() {
```

Comment:
IMO this test should be moved into the `providertest/eventstore` package.

```suggestion
	It("includes event store items in the transaction result", func() {
```